scaleway/resource_lb.go

```diff
@@ -204,17 +282,50 @@ func resourceScalewayLbDelete(ctx context.Context, d *schema.ResourceData, meta
 		return diag.FromErr(err)
 	}
 
-	_, err = lbAPI.WaitForLb(&lb.ZonedAPIWaitForLBRequest{
+	// check if current lb is on stable state
+	retryInterval := DefaultWaitLBRetryInterval
+	currentLB, err := lbAPI.WaitForLb(&lb.ZonedAPIWaitForLBRequest{
 		LBID:          ID,
 		Zone:          zone,
 		Timeout:       scw.TimeDurationPtr(LbWaitForTimeout),
-		RetryInterval: DefaultWaitRetryInterval,
+		RetryInterval: &retryInterval,
 	}, scw.WithContext(ctx))
-
 	if err != nil && !is404Error(err) {
 		return diag.FromErr(err)
 	}
 
+	if currentLB.PrivateNetworkCount != 0 {
```

Comment:
Where do we attach a new private network in case I've added one in a second step? I don't see any attach in this function.